src/efi.rs

```diff
@@ -33,16 +35,47 @@ impl EFI {
         Path::new(MOUNT_PATH).join("EFI")
     }
 
+    fn esp_device(&self) -> PathBuf {
+        Path::new("/dev/disk/by-partlabel/").join(ESP_PART_LABEL)
+    }
+
     fn open_esp_optional(&self) -> Result<Option<openat::Dir>> {
+        self.ensure_mounted_esp()?;
         let sysroot = openat::Dir::open("/")?;
         let esp = sysroot.sub_dir_optional(&self.esp_path())?;
         Ok(esp)
     }
     fn open_esp(&self) -> Result<openat::Dir> {
+        self.ensure_mounted_esp()?;
         let sysroot = openat::Dir::open("/")?;
         let esp = sysroot.sub_dir(&self.esp_path())?;
         Ok(esp)
     }
+
+    fn ensure_mounted_esp(&self) -> Result<()> {
+        let mount_point = &Path::new("/").join(MOUNT_PATH);
```

Comment:
Probably we should create a temporary directory to use as a mountpoint, and then clean it up afterward.

Reply:
We should probably add `PrivateTmp=yes` too.

Reply:
Yeah if we use `PrivateTmp` then we can rely on systemd to clean up our tempdir.
Then I'd use https://docs.rs/tempfile/3.1.0/tempfile/struct.TempDir.html#method.into_path (i.e. make a tempdir, mount on it and then return the path).

Reply:
What do you think about only updating ESP's that are listed in NVRAM? Non-UEFI is trickier, risky to step on all BIOS Boot or MBR gaps with grub-install.

Reply:
For CoreOS, at least, we can make some simplifying assumptions.  We own the whole system, so we can update every ESP and BIOS Boot partition we see.  CoreOS boots via Default Boot Behavior and doesn't set Boot NVRAM variables at all.

If bootupd is going to generalize beyond CoreOS, of course, those assumptions may no longer be workable.

Reply:
Yep, I'm thinking the idea to rebase Silverblue on CoreOS. That use case includes dual boot, so I think "fair game" includes MBR gap, BIOS Boot, and ESP:`/EFI/fedora/` on the drive that contains the current active `/` but not anything else. i.e. EFI/fedora is owned but not the entire ESP. There's low, but non-zero, chance of USB sticks, SD Cards, external drives or even other internal drives, are present and used to boot something else; and also can't be touched.

Reply:
Right, so far bootupd is Fedora CoreOS specific but the idea from the start is it could generalize indeed to other rpm-ostree systems (i.e. not necessarily coreos-assembler generated) as well as even not rpm-ostree systems (traditional yum/dnf/zypper/apt).

It's still not clear to me that the latter effort is really worth it but I don't want to exclude it.